### PR TITLE
feat(data-table): Added data table catalog page

### PR DESCRIPTION
**Pull request by abhiomkar**
Refs https://github.com/material-components/material-components-web/issues/4591

### CODE REVIEW

File: src/DataTableCatalog.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
+import React, {useEffect} from 'react';
```
Comment (moog16) [inline src/DataTableCatalog.js]: Since you're using useEffect, we should explicitly update react to 16.8.x

File: package.json [hunk 2]
```diff
@@ -38,10 +38,11 @@
     "fs-extra": "3.0.1",
     "gh-pages": "^1.1.0",
     "history": "^4.7.2",
+    "htm": "^2.1.1",
```
Comment (moog16) [inline package.json]: what is this for? i thought this was templating

File: src/DataTableCatalog.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  selectedRowIndexes: [1],
+};
+
+const DataTableHero = () => <DataTable data={sampleData} />;
+const DataTableDemos = () => (
+  <div>
+    <h4 className={'demo-data-table-header'}>Data Table Standard</h4>
+    <DataTable data={sampleData} />
+    <h4 className={'demo-data-table-header'}>Data Table with Row Selection</h4>
+    <DataTable data={sampleRowSelectionData} rowSelection={true} />
```
Comment (moog16) [inline src/DataTableCatalog.js]: when a prop is a true boolean, you can omit the `={true}`. 

replace `rowSelection={true}` --> `rowSelecton`

**Comment by abhiomkar** [comment]
Addressed the review comments, ready for re-review @moog16

**Pull request merged**